### PR TITLE
nix: add precommit hooks for formatting

### PR DESCRIPTION
**Pull request by beezow**
This includes formatting git hooks for rust and nix.  These hooks will get loading when loading `nix develop`, and only get ran when you change a file they apply to.  See https://github.com/cachix/pre-commit-hooks.nix for more info on how it works.
Example output when running `git commit`:
```> git commit                                                                       13:28:59
alejandra............................................(no files to check)Skipped
cargo fmt................................................................Failed
- hook id: cargofmt
- exit code: 1

Diff in /home/azoghby/env/humility/cmd/probe/src/lib.rs at line 200:
                     patch.patch_version(),
                 )
             } else {
-
                 format!("<unknown NXP M33 0x{:x}>", coreinfo.manufacturer_part)
             }
         } else {
```

I often find myself forgetting to `cargo fmt` so hopefully this will help prevent having to run `cargo fmt` and rebase.

### CODE REVIEW

**Review by picostove — CHANGES_REQUESTED** [review]
might be good to add as a check so we can catch in CI too?

**Comment by beezow** [comment]
> might be good to add as a check so we can catch in CI too?

`cargo fmt` is already within the checks for building humility, but I will add a nix fmt check.

File: flake.nix
Comment (picostove) [inline flake.nix:51]: hm, looks like this needs to be
```nix
"${pkgs.rustfmt}/bin/cargo-fmt fmt --check --all"
```

Comment (beezow) [reply flake.nix:51]: That is what the default is, but I am trying to force it to use the nightly version of the toolchain.  It works fine when run as a precommit hook. I do not know why it fails when run within the check phase.

Comment (beezow) [reply flake.nix:51]: Switching to using `rustfmt` rather than `cargo fmt` from my custom rust set fixed it.

**Review by picostove — DISMISSED** [review]
lgtm once the fmt discrepancies get resolved

(are these reviews sticky?)

**Comment by beezow** [comment]
Running the `cargo fmt` hook in CI is being problematic and running `rustfmt` causes a cascade of changes in humility.  I switched back to using `cargo fmt` but don't explicitly run the pre-commit checks in CI.  `cargo fmt` is checked within the humility check phase.  
For this reason I also removed the nix fmt hooks as it seems wrong to have a pre-commit hook but no CI gate.

**Comment by beezow** [comment]
This is good to go. I think sorting out the above issues will be more work than it is worth, so I would rather just merge in the cargo fmt hooks now.  We can possible sort something else out later.

**Comment by beezow** [comment]
> lgtm once the fmt discrepancies get resolved
> 
> (are these reviews sticky?)

They are not sticky, but your approval is more symbolic as I need an approval from someone with repository write access.  Since you don't have any problems with the pre-commit-hook itself, ill just merge in once I get another approval.

**Review by FawazTirmizi — APPROVED** [review]

**Pull request merged**
